# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/transportbridge/MediaPipeline.cpp
# Commit: b5228352d20d
# Full Hash: b5228352d20db6c4899ddc3de268ebc35575f0d7
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Fix up AudioConduit audio thread access model, and the channel count for silence. r=padenot
#   
#   This patch uses mMutex to guarantee that the audio thread does not get things
#   pulled away from under its feet. When on the AudioProxyThread we allow blocking
#   the thread during negotiation. On the real audio thread we avoid blocking on
# ==============================================================================

diff -r 056b43129502 -r b5228352d20d dom/media/webrtc/transportbridge/MediaPipeline.cpp
--- a/dom/media/webrtc/transportbridge/MediaPipeline.cpp	Fri Jan 29 15:02:58 2021 +0100
+++ b/dom/media/webrtc/transportbridge/MediaPipeline.cpp	Fri Jan 29 15:53:28 2021 +0100
@@ -1430,7 +1430,8 @@
                  mSource->TrackTimeToSeconds(aDesiredTime)));
         mAudioFrame->UpdateFrame(
             mAudioFrame->timestamp_, nullptr, samplesPer10ms, mRate,
-            mAudioFrame->speech_type_, mAudioFrame->vad_activity_, 1);
+            mAudioFrame->speech_type_, mAudioFrame->vad_activity_,
+            std::max(1UL, mAudioFrame->num_channels()));
       }
 
       MOZ_LOG(
